### ISSUE TITLE
Using localhost as the host can be tricky

### ISSUE THREAD

**Issue by gravesm** [ansible/terraform-provider-ansible#30]
As was highlighted in #28, using `localhost` with this provider can be tricky. It's not obvious to the user why running a playbook with `hosts: localhost` will work when run using `ansible-playbook` directly, but will usually not when the playbook is run with this provider.

Ansible won't use the implicit localhost mechanism when the host specified in the `ansible_playbook` resource and in the playbook is `localhost`, because the provider creates a temporary inventory file with `localhost` in it. The solution is to add `ansible_connection = "local"` to the `extra_vars` block in the TF config, but none of this is obvious.

At the very least, we should add documentation to explain how to properly use localhost, and consider adding support specifically for localhost.

**Comment by framctr** [comment]
I agree to add some documentation on that. Can you better explain which is the solution? Just adding `ansible_connection = "local"` to `extra_vars` seems not to work. I just created a `ansible_playbook` resource with `name = "localhost"` and added `ansible_connection = "local"`. I did not set any `ansible_host` resource for that.

**Comment by Sikhi** [comment]
@framctr 
```
resource "ansible_playbook" "example" {
  playbook                = "simple-playbook.yml"
  name   = "localhost"
  extra_vars = {
    ansible_hostname   = "localhost"
    ansible_connection = "local"
  }
}
```
